Add documentation for hardware-accelerated compression codecs. (#6841)
opensearch-project/documentation-website: _im-plugin/index-codecs.md

When an index segment is created, it uses the current index codec for compression. If you update the index codec, any segment created after the update will use the new compression algorithm. For specific operation considerations, see [Index codec considerations for index operations](#index-codec-considerations-for-index-operations).
{: .note}

> As of OpenSearch 2.14, hardware-accelerated compression codecs for DEFLATE and LZ4 compression algorithms are available. These hardware-accelerated codecs are available on the latest 4th and 5th Gen Intel®️ Xeon®️ processors running Linux kernel 3.10 and later. For all other systems and platforms, the codecs use their corresponding software implementations. 

Reviewer: ```suggestion
As of OpenSearch 2.14, hardware-accelerated compression codecs for the `DEFLATE` and `LZ4` compression algorithms are available. These hardware-accelerated codecs are available on the latest 4th and 5th Gen Intel®️ Xeon®️ processors running Linux kernel 3.10 and later. For all other systems and platforms, the codecs use that platform's corresponding software implementations. 
```

> The new hard-accelerated codecs can be used by setting the value of `index.codec`:

Reviewer: ```suggestion
The new hardware-accelerated codecs can be used by setting one of the following values of the `index.codec` setting:
```

> * `qat_lz4` (OpenSearch 2.14 and later) -- uses hardware-accelerated LZ4. 

Reviewer: ```suggestion
* `qat_lz4` (OpenSearch 2.14 and later): Hardware-accelerated `LZ4`. 
```

> * `qat_deflate` (OpenSearch 2.14 and later)  -- uses hardware-accelerated DEFLATE. 

Reviewer: ```suggestion
* `qat_deflate` (OpenSearch 2.14 and later): Hardware-accelerated `DEFLATE`. 
```

`qat_deflate` offers a much better compression ratio than `qat_lz4` with a modest drop in compression and decompression speed.
{: .note}

> The `index.codec.compression_level` can be used to specify the compression level for both `qat_lz4` and `qat_deflate`. The new setting, `index.codec.qatmode`, controls the behavior of the hardware accelerator. It takes two valid values:

Reviewer: ```suggestion
The `index.codec.compression_level` setting can be used to specify the compression level for both `qat_lz4` and `qat_deflate`. 

The `index.codec.qatmode` setting controls the behavior of the hardware accelerator and uses on of the following values:
```

> * `auto` -- uses hardware-accelerated compression. Fails over to use software if the hardware accelerator is not available.

Reviewer: ```suggestion
* `auto`: If hardware acceleration fails, the algorithm switches to software acceleration.
```

> * `hardware` -- use hardware-accelerated compression. An exception is thrown if hardware is not available (guarantees a hardware-only compression).

Reviewer: ```suggestion
* `hardware`: Guarantees hardware-only compression. If hardware is not available, an exception occurs until hardware exists.
```

> Refer to the snapshot section of this page for the implications `index.codec.qatmode` on snapshot and restore.

Reviewer: ```suggestion
For information about the `index.codec.qatmode` settings implications on snapshots, see the [Snapshots](#snapshots) section.
```

> The hardware accelerator that is used is [Intel (R) QAT accelerator](https://www.intel.com/content/www/us/en/architecture-and-technology/intel-quick-assist-technology-overview.html).

Reviewer: ```suggestion
For more information about hardware acceleration on Intel, see the [Intel (R) QAT accelerator overview](https://www.intel.com/content/www/us/en/architecture-and-technology/intel-quick-assist-technology-overview.html).
```
